File: CS/MainForm.cs
using System;
using DevExpress.XtraCharts;

namespace DateTimeScales {
    public partial class MainForm : DevExpress.XtraEditors.XtraForm {
        public MainForm() {
            InitializeComponent();
        }

        #region #OnAxisScaleChangedExample
        private void Form1_Load(object sender, EventArgs e) {
            this.ordersTableAdapter.Fill(this.nwindDataSet.Orders);
            this.chartControl.AxisScaleChanged += OnAxisScaleChanged;
        }

        private void OnAxisScaleChanged(object sender, AxisScaleChangedEventArgs e) {
            AxisX axis = e.Axis as AxisX;
            DateTimeScaleChangedEventArgs args = e as DateTimeScaleChangedEventArgs;
            if ((args == null) || (args == null)) return;
            axis.Title.Text = String.Format(
                "The Axis Grid Alignment Unit is {0}\r\nThe Axis Measure Unit is {1}", 
                args.GridAlignmentChange.NewValue,
                args.MeasureUnitChange.NewValue);
        }
        #endregion #OnAxisScaleChangedExample
    }
}

File: VB/MainForm.vb
Imports System
Imports DevExpress.XtraCharts

Namespace DateTimeScales
    Partial Public Class MainForm
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        #Region "#OnAxisScaleChangedExample"
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            Me.ordersTableAdapter.Fill(Me.nwindDataSet.Orders)
            AddHandler Me.chartControl.AxisScaleChanged, AddressOf OnAxisScaleChanged
        End Sub

        Private Sub OnAxisScaleChanged(ByVal sender As Object, ByVal e As AxisScaleChangedEventArgs) Handles chartControl.AxisScaleChanged
            Dim axis As AxisX = TryCast(e.Axis, AxisX)
            Dim args As DateTimeScaleChangedEventArgs = TryCast(e, DateTimeScaleChangedEventArgs)
            If (args Is Nothing) OrElse (args Is Nothing) Then
                Return
            End If
            axis.Title.Text = String.Format("The Axis Grid Alignment Unit is {0}" & ControlChars.CrLf & "The Axis Measure Unit is {1}", args.GridAlignmentChange.NewValue, args.MeasureUnitChange.NewValue)
        End Sub
        #End Region ' #OnAxisScaleChangedExample
    End Class
End Namespace
